clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -43,6 +43,54 @@ mlir::Value CIRGenFunction::emitX86BuiltinExpr(unsigned builtinID,
   // Find out if any arguments are required to be integer constant expressions.
   assert(!cir::MissingFeatures::handleBuiltinICEArguments());
 
+  llvm::SmallVector<mlir::Value> ops;
+
+  // Find out if any arguments are required to be integer constant expressions.
+  unsigned iceArguments = 0;
+  ASTContext::GetBuiltinTypeError error;
+  getContext().GetBuiltinType(builtinID, error, &iceArguments);
+  assert(error == ASTContext::GE_None &&
+         "builtinID should be checked before calling emitX86BuiltinExpr");
+  for (auto [idx, arg] : llvm::enumerate(expr->arguments())) {
+    ops.push_back(emitScalarOrConstFoldImmArg(iceArguments, idx, expr));
+  }
+
+  // OG has unordered comparison as a form of optimization in addition to
+  // ordered comparison, while CIR doesn't.
+  //
+  // This means that we can't encode the comparison code of UGT (unordered
+  // greater than), at least not at the CIR level.
+  //
+  // The boolean shouldInvert compensates for this.
+  // For example: to get to the comparison code UGT, we pass in
+  // emitVectorFCmp (OLE, shouldInvert = true) since OLE is the inverse of UGT.
+
+  // There are several ways to support this otherwise:
+  // - register extra CmpOpKind for unordered comparison types and build the
+  // translation code for
+  //    to go from CIR -> LLVM dialect. Notice we get this naturally with
+  //    shouldInvert, benefiting from existing infrastructure, albeit having to
+  //    generate an extra `not` at CIR).
+  // - Just add extra comparison code to a new VecCmpOpKind instead of
+  // cluttering CmpOpKind.
+  // - Add a boolean in VecCmpOp to indicate if it's doing unordered or ordered
+  // comparison
+  // - Just emit the intrinsics call instead of calling this helper, see how the
+  // LLVM lowering handles this.
+  auto emitVectorFCmp = [this, &ops, &expr](cir::CmpOpKind pred,
+                                            bool shouldInvert,
+                                            bool isSignaling) {
+    assert(!cir::MissingFeatures::cgFPOptionsRAII());
+    auto loc = getLoc(expr->getExprLoc());
+    // TODO: Add isSignaling boolean once emitConstrainedFPCall implemented
```

Comment:
```suggestion
    // TODO(cir): Add isSignaling boolean once emitConstrainedFPCall implemented
```
This distinguishes CIR-specific TODO comments from TODO comments we copied from classic codegen.